File: backend/app/schemas/api_document.py
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, ConfigDict, Field


Severity = Literal["high", "medium", "low"]


class ApiDocumentFinding(BaseModel):
    id: str
    title: str
    description: str
    severity: Severity = "medium"
    category: str = "接口文档分析"
    adopted: bool = False


class ApiDocumentBase(BaseModel):
    project_id: Optional[int] = None
    name: str = Field(min_length=1, max_length=255)
    parent_id: Optional[int] = None
    title: Optional[str] = Field(default=None, max_length=255)
    content: Optional[str] = None
    created_by: Optional[str] = Field(default=None, max_length=100)
    is_directory: bool = False
    ai_suggest: list[ApiDocumentFinding] = Field(default_factory=list)


class ApiDocumentCreate(ApiDocumentBase):
    pass


class ApiDocumentUpdate(BaseModel):
    project_id: Optional[int] = None
    name: Optional[str] = Field(default=None, min_length=1, max_length=255)
    parent_id: Optional[int] = None
    title: Optional[str] = Field(default=None, max_length=255)
    content: Optional[str] = None
    created_by: Optional[str] = Field(default=None, max_length=100)
    is_directory: Optional[bool] = None
    ai_suggest: Optional[list[ApiDocumentFinding]] = None


class ApiDocumentOut(ApiDocumentBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None


class ApiDocumentAnalysisRequest(BaseModel):
    document_id: Optional[int] = None
    title: Optional[str] = None
    content: str = Field(min_length=1)
    extra_prompt: Optional[str] = Field(default=None, max_length=1000)


class ApiDocumentAnalysisResponse(BaseModel):
    document_id: Optional[int] = None
    title: Optional[str] = None
    findings: list[ApiDocumentFinding]

File: fronted/src/api/apiDocument.ts
import { del, get, post, put } from './http'
import type { AgentTask } from './agentTask'
import { getPageItems, type PaginatedData } from './pagination'

export type ApiFindingSeverity = 'high' | 'medium' | 'low'

export interface ApiDocumentFinding {
  id: string
  title: string
  description: string
  severity: ApiFindingSeverity
  category: string
  adopted: boolean
}

export interface ApiDocument {
  id: number
  project_id?: number | null
  name: string
  parent_id?: number | null
  title?: string | null
  content?: string | null
  created_by?: string | null
  is_directory: boolean
  ai_suggest: ApiDocumentFinding[]
  created_at?: string | null
  updated_at?: string | null
}

export interface ApiDocumentPayload {
  project_id?: number | null
  name: string
  parent_id?: number | null
  title?: string | null
  content?: string | null
  created_by?: string | null
  is_directory?: boolean
  ai_suggest?: ApiDocumentFinding[]
}

export interface ApiDocumentAnalysisRequest {
  document_id?: number
  title?: string | null
  content: string
  extra_prompt?: string
}

export interface ApiDocumentAnalysisResponse {
  document_id?: number | null
  title?: string | null
  findings: ApiDocumentFinding[]
}

export function getApiDocuments(projectId?: number): Promise<ApiDocument[]> {
  return get<PaginatedData<ApiDocument>>(
    '/api/v1/api-documents/',
    projectId ? { project_id: projectId } : undefined,
  ).then(getPageItems)
}

export function createApiDocument(data: ApiDocumentPayload): Promise<ApiDocument> {
  return post<ApiDocument>('/api/v1/api-documents/', data)
}

export function updateApiDocument(
  documentId: number,
  data: Partial<ApiDocumentPayload>,
): Promise<ApiDocument> {
  return put<ApiDocument>(`/api/v1/api-documents/${documentId}`, data)
}

export function deleteApiDocument(documentId: number): Promise<ApiDocument> {
  return del<ApiDocument>(`/api/v1/api-documents/${documentId}`)
}

export function analyzeApiDocument(
  data: ApiDocumentAnalysisRequest,
): Promise<AgentTask<ApiDocumentAnalysisResponse>> {
  return post<AgentTask<ApiDocumentAnalysisResponse>>('/api/v1/api-documents/analysis', data)
}
